app/(private-pages)/admin-dashboard/package/_components/singular-package-to-edit.tsx

```diff
@@ -0,0 +1,301 @@
+'use client';
+
+import { useState } from 'react';
+import { motion, AnimatePresence } from 'framer-motion';
+import { Formik, Form } from 'formik';
+import * as Yup from 'yup';
+import { BasicDetailsForm } from './new-package/basic-details';
+import { PackageDetailsForm } from './new-package/package-details';
+import { CategoryForm } from './new-package/category-form';
+import AppButton from '@/components/reusables/AppButton';
+import { FaArrowLeft, FaArrowRight } from 'react-icons/fa';
+import { useAppToast } from '@/components/reusables/AppToast'
+import { useRouter } from 'next/navigation';
+import AppDialogBox from '@/components/reusables/AppDialogBox';
+import { EditPackageProps, Package } from '@/constants/types';
+import { PaginatedResponse } from '@/api/types';
+import { segregatePackageByItsPriceCategory } from '@/lib/utils';
+import { useEditPackage, useGetAllActivePackagesById } from '@/api/services/packages';
+
+const validationSchema = Yup.object().shape({
+  name: Yup.string().required('Package name is required'),
+  description: Yup.string().required('Package description is required'),
+  package_type: Yup.string().required('Package type is required'),
+  expiry_date: Yup.string().required('Expiry date is required'),
+  package_prices: Yup.array().of(
+    Yup.object().shape({
+      category: Yup.string().required(),
+      price: Yup.string().required('Price is required')
+    })
+  ).required('Package prices are required'),
+  category_descriptions: Yup.array().of(
+    Yup.object().shape({
+      category: Yup.string().required(),
+      description: Yup.string().required('Category description is required')
+    })
+  ).required('Category descriptions are required')
+});
+
+const initialValues = {
+  name: '',
+  description: '',
+  package_type: '',
+  expiry_date: '',
+  package_prices: [
+    { category: 'vip', price: '' },
+    { category: 'deluxe', price: '' },
+    { category: 'standard', price: '' }
+  ],
+  category_descriptions: [
+    { category: 'vip', description: '' },
+    { category: 'deluxe', description: '' },
+    { category: 'standard', description: '' }
+  ],
+  is_active: true
+};
+
+export default function SingularPackageToEdit({ packages: packageData, id}: { packages: PaginatedResponse<Package>, id: string }) {
```

Comment:
this is not done properly. you are trying to view a singularpackage. what is the reason behind making a paginated response of all the active package as its props. shouldn't it be that only a singular package instance is whats passed here